### ISSUE TITLE
[Accessibility] [Modal] Content is not within a landmark element

### ISSUE THREAD

**Issue by diego-king** [carbon-design-system/carbon#8978]
## Environment

> Operating system
Windows

> Browser
Chrome 91.0.4472.106

> Assistive technology used to verify
IBM Equal Access Accessibility Checker Version 3.1.8

## Detailed description

> What version of the Carbon Design System are you using?
7.25.0

> What did you expect to happen?
No violation.

> What happened instead?
A violation occurred.
![image](https://user-images.githubusercontent.com/32442955/122820692-e3e9fa00-d2a9-11eb-8365-946b71757719.png)

> What WCAG 2.1 checkpoint does the issue violate?
WCAG 2.1 technique ARIA11

## Steps to reproduce the issue

1. Step one: Open a modal.
2. Step two: Use IBM Equal Access Accessibility Checker to scan.
3. Step three: This violation is detected.

### Element location
`<span class="bx--visually-hidden" role="link" tabindex="0">`

### What to do
Enclose the content within an HTML5 sectioning element with an implicit landmark role (e.g. `<main>Content</main>`);
OR use the "role" attribute on an enclosing element to specify an appropriate WAI-ARIA landmark value (e.g. `<div role="contentinfo">...</div>`).

**Comment by joshblack** [comment]
Hi there! 👋  This violation does not apply to modal dialogs and is incorrectly being raised by Accessbility Checker, I believe.

The implementation itself for Modal will likely get a refresh, but for now you can safely disregard this report from AC 👍

**Comment by joshblack** [comment]
Just wanted to follow up, if you're an IBMer here is the issue on the IBMa side: https://github.ibm.com/IBMa/E2E/issues/3276

**Comment by tombrunet** [comment]
@diego-king Do you have an example of where this is failing (e.g., in the React storybook?). We dug into the code and dialogs should have already been ignored in the current rulesets.
